### PR TITLE
Add main functionality

### PR DESCRIPTION
**Pull request by smirnovanv**

### CODE REVIEW

File: queue.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Queue {
+    constructor() {
+        this.data = [];
```
Comment (m0sk1t) [inline queue.js]: data shoud be private member

File: stack.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Stack {
+    constructor() {
+        this.data = [];
```
Comment (m0sk1t) [inline stack.js]: data shoud be private member

File: strangeQueue.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+        this.stackOne.add(dataUnit); 
+    }
+
+    remove() {
+        while (this.stackOne.data.length !== 0) {
+            const currentUnit = this.stackOne.remove();
+            this.stackTwo.add(currentUnit);
+        }
+        this.stackTwo.remove();
```
Comment (m0sk1t) [inline strangeQueue.js]: where is return?)

File: strangeQueue.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { Stack } from './stack.js';
+
+class StrangeQueue {
+    constructor() {
+        this.stackOne = new Stack();
+        this.stackTwo = new Stack();
+    }
+
+    add(dataUnit) {
+        while (this.stackTwo.data.length !== 0) {
```
Comment (m0sk1t) [inline strangeQueue.js]: data should be private member
try to use something like

```js
while (const currentUnit = this.stackTwo.remove()) {
...
}
```

**Review by m0sk1t — CHANGES_REQUESTED** [review]
LGTM) Great job! Just small suggestions

File: index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { Queue } from './queue.js';
+import { StrangeQueue } from './strangeQueue.js';
```
Comment (m0sk1t) [inline index.js]: `StrangeQueue` naming gives me zero information about this structure, consider to use smthng like `TwoStackQueue`

File: index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+const queue = new Queue();
+
+queue.add('item 1');
+
```
Comment (m0sk1t) [inline index.js]: too many new lines)

File: queue.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    }
+
+    add(dataUnit) {
+        this.data.push(dataUnit);
+    }
+
+    remove() {
+        this.data.shift();
```
Comment (m0sk1t) [inline queue.js]: return?

File: queue.js (hunk 1 above)
Comment (smirnovanv) [reply queue.js]: done

File: stack.js (hunk 2 above)
Comment (smirnovanv) [reply stack.js]: done

File: strangeQueue.js (hunk 3 above)
Comment (smirnovanv) [reply strangeQueue.js]: done

File: strangeQueue.js (hunk 4 above)
Comment (smirnovanv) [reply strangeQueue.js]: changed the method

File: index.js (hunk 5 above)
Comment (smirnovanv) [reply index.js]: done

File: index.js (hunk 6 above)
Comment (smirnovanv) [reply index.js]: removed

File: queue.js (hunk 7 above)
Comment (smirnovanv) [reply queue.js]: done

File: index.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+queue.add('item 1');
+queue.add('item 2');
+queue.add('item 3');
+queue.remove();
+queue.add('item 4');
+
+console.log(queue.showStatus());
+
+const strange = new TwoStackQueue();
```
Comment (m0sk1t) [inline index.js]: strange? Doctor Strange? don't forget to rename all related code, because now it looks strange 🤣

File: queue.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+    constructor() {
+        this.#data = [];
+    }
+
+    add(dataUnit) {
+        this.#data.push(dataUnit);
+    }
+
+    remove() {
```
Comment (m0sk1t) [inline queue.js]: maybe `get` or `retrieve` or `unqueue` or `pop`, but it is just my assumption, not a requirement

**Review by m0sk1t — CHANGES_REQUESTED** [review]

File: index.js (hunk 8 above)
Comment (smirnovanv) [reply index.js]: done

File: queue.js (hunk 9 above)
Comment (smirnovanv) [reply queue.js]: done

File: strangeQueue.js (hunk 4 above)
Comment (smirnovanv) [reply strangeQueue.js]: fixed

**Review by m0sk1t — APPROVED** [review]

**Pull request merged**